### PR TITLE
Update use cases

### PR DESCRIPTION
**Pull request by myang2020**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -616,63 +626,111 @@ A successful addition adds a person to the global participant list.
...
 
+**Guarantees:**
+A successful removal removes the person from the selected event.
+
 **MSS:**
-1. User requests to remove a participant from a certain event.
-2. Eventy remove the participant from the event. </br>
+1. User has an event selected.
+2. User requests to remove a participant from a certain event.
+3. Eventy remove the participant from the event. </br>
```
Comment (ChocoRagdoll) [inline docs/DeveloperGuide.md:693]: removes

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -686,37 +744,203 @@ A successful selection displays the event details.
...
+
+**Actor:** User
+
+**Preconditions:** User has added one or more contacts to the global contact list or invited one or more contacts to the selected event.
+
+**Guarantees:**
+A successful export exports target contacts or invitees to a csv file.
+
+**MSS:**
+1. User enters tags of the contacts or invitees he wants to export.
```
Comment (ChocoRagdoll) [inline docs/DeveloperGuide.md:783]: tags of -> flags to specify the details of

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -686,37 +744,203 @@ A successful selection displays the event details.
...
+2. Eventy exports the contacts or invitees along with the specified tags to a csv file </br>
```
Comment (ChocoRagdoll) [inline docs/DeveloperGuide.md:784]: exports the specified details of the contacts or invitees to a csv file

**Review by ChocoRagdoll — COMMENTED** [review]
only some minor changes neccessary

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -530,14 +530,24 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+| `* *`    | user                                       | delete an event               | remove past or irrelevant events                                       |
+| `* * *`  | user                                       | add a new person              | manage all my contacts in one place                                    |
+| `* * *`  | user                                       | invite participant to an event| manage and plan all the participants for a specific event              |
+| `* * *`  | user                                       | delete a person               | remove contacts that I no longer need                                  |
+| `* * *`  | user                                       | remove specific participant   | remove a specific participant from a specific event                    |
+| `* * *`  | diligent user                              | edit contact number           | update contact number                                                  |
+| `* *`    | user                                       | select a specific event       | Perform actions on the desired event                                   |
```
Comment (Apzure) [inline docs/DeveloperGuide.md:540]: The benefit for this doesn't sound concrete

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -530,14 +530,24 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+| `* *`    | organized user                             | filter contacts by tags       | view the list of contacts or invitees in a specific category           |
+| `* *`    | organized user                             | export filtered contacts      | quickly view consolidated information in CSV file format              |
+| `* *`    | user                                       | edit an event                 | update event information like the event name or event date             |
+| `* *`    | user                                       | deselect an event             | perform app-wide events                                                |
```
Comment (Apzure) [inline docs/DeveloperGuide.md:544]: same for this

File: docs/DeveloperGuide.md [hunk 6]
```diff
@@ -686,37 +744,203 @@ A successful selection displays the event details.
...
+1. User enters index of contact he wants to edit, along with the new tags.
+2. Eventy updates the tags of the contact. </br>
+   *Use case ends*
+
+**Extensions:**
+
+1a. The participant index provided by the User is invalid.
+- 1a1. Eventy displays an error message to alert the User. </br>
+  *Use case ends*
+
```
Comment (Apzure) [inline docs/DeveloperGuide.md:943]: Having one for each field seems excessive, could you try thinking of new ones? There can be use cases with more depth.

File: docs/DeveloperGuide.md [hunk 7]
```diff
@@ -686,37 +744,203 @@ A successful selection displays the event details.
...
 
 **Guarantees:**
-A successful addition adds a participant from the global participant list to the selected event.
+Eventy displays the contacts with target tags.
 
 **MSS:**
-1. User inputs the index of participant to be added.
-2. Eventy displays the added participant in the panel showing list of participants in selected event. </br>
+
+1. User selects an event or remains in global contact list.
```
Comment (Apzure) [inline docs/DeveloperGuide.md:760]: are you allowed to have this kind of OR in use cases?

**Review by Apzure — DISMISSED** [review]

File: docs/DeveloperGuide.md [hunk 8]
```diff
@@ -616,63 +627,111 @@ A successful addition adds a person to the global participant list.
...
+**Actor:** User
 
-1b. Participant index provided by the User is invalid.
+**Preconditions:** User has added one or more participant to the global contact list.
+
+**Guarantees:**
+A successful edit updates the person's phone number in both the global contact list and all the events he participates in.
+
+**MSS:**
+1. User enters index of contact he wants to edit, along with the new phone number.
```
[the remark this answers is not in the document]
Comment (nusliuyifan) [reply docs/DeveloperGuide.md:715]: Oh, I just realised by contact you mean participant, can just ignore this

File: docs/DeveloperGuide.md [hunk 9]
```diff
@@ -686,37 +745,227 @@ A successful selection displays the event details.
...
+### Edit contact email
+
+**Use case: UC16 - Edit email of contact**
+
+**Actor:** User
+
+**Preconditions:** User has added one or more participant to the global contact list.
+
+**Guarantees:**
+A successful edit updates the person's email in both the global contact list and all the events he participates in.
```
Comment (nusliuyifan) [inline docs/DeveloperGuide.md:888]: maybe can use more general pronoun instead of "he", maybe "this person"

Comment (nusliuyifan) [reply docs/DeveloperGuide.md:888]: Can ignore this if there's too much work

**Review by nusliuyifan — CHANGES_REQUESTED** [review]
You can change or just ignore  them as you like, because I'm also not sure about some of them. Please let me know when you need me to approve the pull request.

File: docs/DeveloperGuide.md [hunk 10]
```diff
@@ -686,37 +745,227 @@ A successful selection displays the event details.
...
 **Guarantees:**
-A successful addition adds a participant from the global participant list to the selected event.
+Eventy displays the contacts with target tags.
 
 **MSS:**
-1. User inputs the index of participant to be added.
-2. Eventy displays the added participant in the panel showing list of participants in selected event. </br>
+
+1. User selects an event or remains in global contact list.
+2. User specific the tags he wants to filter by.
```
Comment (taeewonnn) [inline docs/DeveloperGuide.md:762]: specifies

**Review by taeewonnn — CHANGES_REQUESTED** [review]
rest lgtm

**Review by taeewonnn — APPROVED** [review]
lgtm

**Pull request merged**